Set focus to skip link target to improve screen reader announcements (#2450)
alphagov/govuk-frontend: CHANGELOG.md

This change was introduced in [pull request #2408: Prevent issues with character count when textarea `id` includes CSS syntax characters](https://github.com/alphagov/govuk-frontend/pull/2408).

> #### Include JavaScript for the skip link to improve screen reader announcements

Reviewer: ```suggestion
#### Include JavaScript for skip link to improve screen reader announcements
```

> We've added JavaScript for the skip link component to set focus to the target element. This improves the announcements especially for Mac VoiceOver users and also for JAWS users.

Reviewer: Should we give an example of what we mean by 'target element' (if there could be any inexperienced users who wouldn't understand this right away)? "The main content on a page"?

How does it improve the announcements? By reading the main element when users activate the skip link - which is different to what happened before we made this change?

Last comment: we had some feedback from release notes testing that some of our users are junior devs who are inexperienced when it comes to accessibility. Do we need to spell out the JAWS acronym for them, or provide any links?

```suggestion
We've added JavaScript for the skip link component to set focus to the target element, for example, the main content on a page. This change improves the announcements, especially for users of Mac VoiceOver and JAWS.
```

Author: Thanks for the feedback Eoin, all really good suggestions. I've gone with something that's quite close to what you suggested including removing some of the jargon. Have a look and let me know if you think it needs further tweaking.

> We've added JavaScript for the skip link component to set focus to the linked element, for example, the main content on the page. This helps screen readers read the linked content when users activate the skip link.

Reviewer: ```suggestion
We've added JavaScript for the skip link component to set focus to the linked element, for example, the main content on the page. This helps screen readers read the linked content when users turn on the skip link.
```

> If you are not using Nunjucks macros, add a `data-module="govuk-skip-link"` attribute to the component HTML. For example:

Reviewer: ```suggestion
If you're not using Nunjucks macros, add a `data-module="govuk-skip-link"` attribute to the component HTML. For example:
```

```html
<div class="govuk-skip-link" data-module="govuk-skip-link">
...
</div>
```

If you're [importing JavaScript for individual components](https://frontend.design-system.service.gov.uk/importing-css-assets-and-javascript/#select-and-initialise-an-individual-component), import the skip link JavaScript.

> Once you've made the changes, you can check that the skip link works correctly by clicking it and   inspecting the target element (commonly the `<main>` element) in the browser to ensure it has a `tabindex` attribute.

Reviewer: ```suggestion
Once you've made the changes, check that the skip link works. To make sure, click the skip link and check that the target element (usually the `<main>` element) in the browser has a `tabindex` attribute.
```

Author: Thanks @EoinShaughnessy, this is a nice improvement 👍  

I've gone with your wording but I've added the word 'JavaScript'. This is because strictly speaking we want to them to check that the skip link _JavaScript_ works. You might get the impression that the skip link has worked since the browser has shifted the page content but the JavaScript might have not fired - checking for the `tabindex` ensures that it did.

> This change was introduced in [pull request #2450: ](https://github.com/alphagov/govuk-frontend/pull/2450).

Reviewer 2: Missing PR title?

Reviewer: ```suggestion
This change was introduced in [pull request #2450: Set focus to skip link target to improve screen reader announcements](https://github.com/alphagov/govuk-frontend/pull/2450).
```
